File: 199-binary-tree-right-side-view/199-binary-tree-right-side-view.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */
class Solution {
    public List<Integer> rightSideView(TreeNode root) {
        ArrayList<Integer> ans = new ArrayList<>();
        
        if(root == null)
            return ans;
        
        ArrayList<TreeNode> q = new ArrayList<>();
        q.add(root);
        
        while(q.size() > 0){
            ans.add(q.get(q.size() - 1).val);
            
            ArrayList<TreeNode> nq = new ArrayList<>();
            for(int i = 0; i < q.size(); i++){
                TreeNode curr = q.get(i);
                if(curr.left != null)
                    nq.add(curr.left);
                if(curr.right != null)
                    nq.add(curr.right);
            }
            q = nq;
        }
        
        return ans;
    }
}

File: 199-binary-tree-right-side-view/199-binary-tree-right-side-view.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    def rightSideView(self, root: Optional[TreeNode]) -> List[int]:
        if not root:
            return []
        
        q = [root]
        ans = []
        while q:
            ans.append(q[-1].val)
            
            nq = []
            for curr in q:
                if curr.left:
                    nq.append(curr.left)
                if curr.right:
                    nq.append(curr.right)
            q = nq
        return ans